lib/game/widget_builder.dart

```diff
@@ -0,0 +1,202 @@
+      onForcePressPeak: game is ForcePressDetector
+          ? (ForcePressDetails d) =>
+              (game as ForcePressDetector).onForcePressPeak(d)
+          : null,
+      onForcePressUpdate: game is ForcePressDetector
+          ? (ForcePressDetails d) =>
+              (game as ForcePressDetector).onForcePressUpdate(d)
+          : null,
+      onForcePressEnd: game is ForcePressDetector
+          ? (ForcePressDetails d) =>
+              (game as ForcePressDetector).onForcePressEnd(d)
+          : null,
+
+      // Pan
+      onPanDown: game is PanDetector
+          ? (DragDownDetails d) => (game as PanDetector).onPanDown(d)
+          : null,
+      onPanStart: game is PanDetector
+          ? (DragStartDetails d) => (game as PanDetector).onPanStart(d)
+          : null,
+      onPanUpdate: game is PanDetector
+          ? (DragUpdateDetails d) => (game as PanDetector).onPanUpdate(d)
+          : null,
+      onPanEnd: game is PanDetector
+          ? (DragEndDetails d) => (game as PanDetector).onPanEnd(d)
+          : null,
+      onPanCancel: game is PanDetector
+          ? () => (game as PanDetector).onPanCancel()
+          : null,
+
+      // Scales
+      onScaleStart: game is ScaleDetector
+          ? (ScaleStartDetails d) => (game as ScaleDetector).onScaleStart(d)
+          : null,
+      onScaleUpdate: game is ScaleDetector
+          ? (ScaleUpdateDetails d) => (game as ScaleDetector).onScaleUpdate(d)
+          : null,
+      onScaleEnd: game is ScaleDetector
+          ? (ScaleEndDetails d) => (game as ScaleDetector).onScaleEnd(d)
+          : null,
+
+      child: Container(
+          color: game.backgroundColor(),
+          child: Directionality(
+              textDirection: TextDirection.ltr,
+              child: EmbeddedGameWidget(game))),
+    );
+  }
+}
+
+class OverlayGameWidget extends StatefulWidget {
+  final Widget child;
+  final HasWidgetsOverlay game;
+
+  OverlayGameWidget({this.child, this.game});
+
+  @override
+  State<StatefulWidget> createState() => _OverlayGameWidgetState(game);
+}
+
+class _OverlayGameWidgetState extends State<OverlayGameWidget> {
+  final Map<String, Widget> _overlays = {};
+
+  _OverlayGameWidgetState(HasWidgetsOverlay game) {
+    game.widgetOverlayController.stream.listen((overlay) {
+      setState(() {
+        if (overlay.widget == null) {
+          _overlays.remove(overlay.name);
+        } else {
+          _overlays[overlay.name] = overlay.widget;
+        }
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Directionality(
+        textDirection: TextDirection.ltr,
+        child: Stack(
+            children: [
+          widget.child,
+        ]..addAll(_overlays.values.toList())));
+  }
+}
+
+class OverlayWidgetBuilder extends WidgetBuilder {
+  OverlayWidgetBuilder();
+
+  @override
+  Widget build(Game game) {
+    final container = super.build(game);
+
+    return OverlayGameWidget(child: container, game: game);
```

Comment:
I think this `child` name misleading since it comes from the superclass and differently, than "child" on normal flutter widgets, this cannot be any kind of widget, so, since it is expected to be a game's widget, I think we can name it accordingly.